node/src/Context.cpp

```diff
@@ -33,13 +35,16 @@ namespace node {
                         .ThrowAsJavaScriptException();
                     return;
                 }
-                std::string powerPreferenceOption = optionsObject.Get("powerPreference").ToString();
-                if (powerPreferenceOption == "default") {
-                    options.powerPreference = MLPowerPreference_Default;
-                } else if (powerPreferenceOption == "low-power") {
-                    options.powerPreference = MLPowerPreference_Low_power;
-                } else if (powerPreferenceOption == "high-performance") {
-                    options.powerPreference = MLPowerPreference_High_performance;
+                std::string powerPreference = optionsObject.Get("powerPreference").ToString();
+                std::transform(powerPreference.begin(), powerPreference.end(),
+                               powerPreference.begin(),
+                               [](char c) -> char { return std::tolower(c); });
```

Comment:
I don't think the implementation should transform them to lowercase. If the string doesn't match the spec definition, it is a wrong type. If it is not set (or undefined), the implementation should use the default value.

Reply:
OK, thanks, I will remove the lowercase.